### PR TITLE
Add Docusaurus knowledgeBase searchbar

### PR DESCRIPTION
**Pull request by vineethasok**

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
This is an automated comment for commit f8f1dedefb0a5bd221f7cb596ae80b82071cf333 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/1125/f8f1dedefb0a5bd221f7cb596ae80b82071cf333/ci_running.html)
The overall status of the commit is 🟢 success

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🟢 success</td></tr>
<tr><td>Vercel</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>🟢 success</td></tr>
</table>

**Comment by vineethasok** [comment]
https://clickhouse-docs-git-193-add-search-bar-to-kno-1f44db-clickhouse.vercel.app/docs/en/intro

File: src/theme/BlogSidebar/Mobile/index.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+              {item.title}
+            </Link>
+          </li>
+        ))}
+      </ul>
+    </>
+  );
+}
+
+export default function BlogSidebarMobile(props) {
```
Comment (guykoh) [inline src/theme/BlogSidebar/Mobile/index.js:30]: What's the reason to create separate components? One for Desktop and one for mobile? Why not merging them into one component? Is this the common solution for supporting mobile?

Comment (vineethasok) [reply src/theme/BlogSidebar/Mobile/index.js:30]: This is exported from the docusaurus side
We use sizzle to get the code from docusaurus end and modify it the rest would be automatically added from the original end
https://docusaurus.io/docs/swizzling

**Review by guykoh — APPROVED** [review]

**Review by gjones — COMMENTED** [review]
These classnames don't seem to be defined anywhere. Separately we have a `bootstrap.css` and `bootstrap.js` file which don't seem to be doing much. Should we create a standalone PR that cleans a lot of this stuff up? cc @justindeguzman

**Review by gjones — APPROVED** [review]
Yeah overall it works as expected. We can fix up the styles shortly.

**Pull request merged**